src/filter.ts

```diff
@@ -0,0 +1,36 @@
+import {Changes} from './schemas'
+import {Severity, SEVERITIES} from './schemas'
+
+export function filterChangesBySeverity(
+  severity: Severity,
+  changes: Changes
+): Changes {
+  const severityIdx = SEVERITIES.indexOf(severity)
+  let filteredChanges = []
+  for (let change of changes) {
+    if (
+      change === undefined ||
+      change.vulnerabilities === undefined ||
+      change.vulnerabilities.length === 0
+    ) {
+      continue
+    }
+
+    let fChange = {
+      ...change,
+      vulnerabilities: change.vulnerabilities.filter(vuln => {
+        const vulnIdx = SEVERITIES.indexOf(vuln.severity)
+        if (vulnIdx <= severityIdx) {
+          return true
```

Comment:
Do we want to fail as soon as we go over the threshold or provide detail of all packages that don't meet the requirement? My only concern is these PR vulnerabilities won't be in security tab of the repo so you won't have visibility for rest of the packages you need to fix.

Reply:
Ah sorry, ignore my comment, I lost track while playing with this branch action. It does print out all packages you need to fix up to the severity level you specified. I mistakenly thought it will fail after the first one.